### PR TITLE
[WIP] Mac OS support

### LINKED ISSUE #166: Support for macOS
Hi,

Are you planning support macOS?

Thanks.

### PR DESCRIPTION
**Pull request by EndrII**
fix #166 
- [ ] Adaprive the [macdependency](https://github.com/QuasarApp/macdependency) project for crosplatform build 
- [ ] Add support macdependency library in the CQtDeployer
- [ ] Add support mac os on tests 
- [x] Add CI Mac OS workers
- [ ] Prepare Builds of The CQtDeployer for the Mac OS 
- [ ] Update documentations

### CODE REVIEW

**Comment by EndrII** [comment]
I can not build the macdependency library. it is impossible for the linux and windows platforms, 
but i thin is good solution move the cqtdeployer to the [LIEF]( https://github.com/lief-project/LIEF) library

**Comment by EndrII** [comment]
###  Work plan

We need to change a current build system to the cmake before include the LIEF library because include the LIEF library using the qmake build system is impossible. #113

1. Add support cmake #113 
2. Use the LIEF library.
3. Add support mac os. #166

File: Deploy/Parsers/macho_type.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
```
Comment (marcelpetrick) [inline Deploy/Parsers/macho_type.cpp:2]: Maybe adjust the date with 2022?

**Comment by EndrII** [comment]
This PR is invalid anymore.

**Pull request closed without merging**